pkg/controllers/clusterresourceplacement/resource_selector.go

```diff
@@ -353,6 +354,14 @@ func generateRawContent(object *unstructured.Unstructured) ([]byte, error) {
 		if err != nil {
 			return nil, fmt.Errorf("failed to get the ports field in Serivce object, name =%s: %w", object.GetName(), err)
 		}
+	} else if object.GetKind() == "Job" && object.GetAPIVersion() == batchv1.SchemeGroupVersion.String() {
+		if manualSelector, exist, _ := unstructured.NestedString(object.Object, "spec", "manualSelector"); !exist || manualSelector == "true" {
+			// remove the selector field added by the api-server if the job is not created with manual selector
+			// https://github.com/kubernetes/kubernetes/blob/d4fde1e92a83cb533ae63b3abe9d49f08efb7a2f/pkg/registry/batch/job/strategy.go#L219
+			unstructured.RemoveNestedField(object.Object, "spec", "selector")
+			unstructured.RemoveNestedField(object.Object, "spec", "template", "metadata", "creationTimestamp")
+			unstructured.RemoveNestedField(object.Object, "spec", "template", "metadata", "labels")
```

Comment:
This clears out all the labels on the pod template, not just the UID label set by the system, which could have some side effects on the user-side, e.g. the user may still want all the child pods to have some labels that are not used for job selectors -> though I am not 100% sure how many users would have such labels

Reply:
echo chen's comment. Can we only clear the the controller-uid  when its value = the job uuid? same for the  job-name.